sycl/include/CL/sycl/detail/common.hpp

```diff
@@ -168,7 +117,7 @@ template <class Obj> decltype(Obj::impl) getSyclObjImpl(const Obj &SyclObject) {
 // must make sure the returned pointer is not captured in a field or otherwise
 // stored - i.e. must live only as on-stack value.
 template <class T>
-typename detail::add_pointer_t<typename decltype(T::impl)::element_type>
+typename std::add_pointer_t<typename decltype(T::impl)::element_type>
```

Comment:
Isn't it a part of another PR?

Reply:
Right, a few things might have slipped in here. It was hard to split it cleanly.